### DISCUSSION TITLE
Route Rules - Redirection to routes with dynamic route parameter(s)

### DISCUSSION THREAD

**Discussion by gomorizsolt** [nuxt/nuxt#28271] [Questions]
The `routeRules` option within the Nuxt config file allows for defining redirects:

```js
export default defineNuxtConfig({
  routeRules: {
    '/old-page': { 
      redirect: '/new-page'
    }
  }
})
```

In my case, I need to support a legacy URL with dynamic route parameters:

```js
export default defineNuxtConfig({
  routeRules: {
    '/foo/:bar': {
      redirect: '/new-page'
    }
  }
})
```

The question is, is there a way to port the dynamic route parameters over to the route the user is about to be redirected to?

```js
export default defineNuxtConfig({
  routeRules: {
    '/foo/:bar': {
      redirect: '/new-page/:bar'
    }
  }
})
```

At the moment, as a workaround the redirection takes place on page level with the `redirect` option in `definePageMeta`:

```html
<script setup lang="ts">
// /pages/foo/[bar].vue
definePageMeta({
  redirect: (to) => {
    return {
      path: '/new-page/:bar',

      params: {
        bar: to.params.bar
      }
    };
  },
});
</script>

```

**Comment by Davvii1** [comment]
Hey! You can use `*` or `**` right after your `/old-page` route as a wildcard to take all of the paths that follow, so you will have something like this:
```ts
export default defineNuxtConfig({
  compatibilityDate: '2024-04-03',
  devtools: { enabled: true },
  routeRules: {
    '/old-routes/**': {
      redirect: '/new-route',
    },
  },
});
```
You can see more examples in nitro config documentation here: [Nitro Config](https://nitro.unjs.io/config#routerules)
and a working example that I made here in stackblitz: [Example](https://stackblitz.com/edit/github-xj8qfv?file=nuxt.config.ts)

**Reply by TO-koswald** [reply]
This is not quite what was asked though, as your example would lose the dynamic parameters. I think what @gomorizsolt needs are wildcard redirects which are supported from nitro 2.9.0:
```
export default defineNuxtConfig({
  routeRules: {
    '/old-routes/**': {
      redirect: '/new-route/**',
    },
  },
});
```
